src/solvers/bitwuzla/bitwuzla_conv.cpp

```diff
@@ -0,0 +1,920 @@
+  assert(b->sort->id != SMT_SORT_INT && b->sort->id != SMT_SORT_REAL);
+  assert(a->sort->get_data_width() == b->sort->get_data_width());
+  return new_ast(
+    bitwuzla_mk_term2(
+      bitw,
+      BITWUZLA_KIND_BV_UGE,
+      to_solver_smt_ast<bitw_smt_ast>(a)->a,
+      to_solver_smt_ast<bitw_smt_ast>(b)->a),
+    boolean_sort);
+}
+
+smt_astt bitwuzla_convt::mk_bvsge(smt_astt a, smt_astt b)
+{
+  assert(a->sort->id != SMT_SORT_INT && a->sort->id != SMT_SORT_REAL);
+  assert(b->sort->id != SMT_SORT_INT && b->sort->id != SMT_SORT_REAL);
+  assert(a->sort->get_data_width() == b->sort->get_data_width());
+  return new_ast(
+    bitwuzla_mk_term2(
+      bitw,
+      BITWUZLA_KIND_BV_SGE,
+      to_solver_smt_ast<bitw_smt_ast>(a)->a,
+      to_solver_smt_ast<bitw_smt_ast>(b)->a),
+    boolean_sort);
+}
+
+smt_astt bitwuzla_convt::mk_eq(smt_astt a, smt_astt b)
+{
+  assert(a->sort->get_data_width() == b->sort->get_data_width());
+  return new_ast(
+    bitwuzla_mk_term2(
+      bitw,
+      BITWUZLA_KIND_EQUAL,
+      to_solver_smt_ast<bitw_smt_ast>(a)->a,
+      to_solver_smt_ast<bitw_smt_ast>(b)->a),
+    boolean_sort);
+}
+
+smt_astt bitwuzla_convt::mk_neq(smt_astt a, smt_astt b)
+{
+  assert(a->sort->get_data_width() == b->sort->get_data_width());
+  return new_ast(
+    bitwuzla_mk_term2(
+      bitw,
+      BITWUZLA_KIND_DISTINCT,
+      to_solver_smt_ast<bitw_smt_ast>(a)->a,
+      to_solver_smt_ast<bitw_smt_ast>(b)->a),
+    boolean_sort);
+}
+
+smt_astt bitwuzla_convt::mk_store(smt_astt a, smt_astt b, smt_astt c)
+{
+  assert(a->sort->id == SMT_SORT_ARRAY);
+  assert(a->sort->get_domain_width() == b->sort->get_data_width());
+  assert(
+    a->sort->get_range_sort()->get_data_width() == c->sort->get_data_width());
+  return new_ast(
+    bitwuzla_mk_term3(
+      bitw,
+      BITWUZLA_KIND_ARRAY_STORE,
+      to_solver_smt_ast<bitw_smt_ast>(a)->a,
+      to_solver_smt_ast<bitw_smt_ast>(b)->a,
+      to_solver_smt_ast<bitw_smt_ast>(c)->a),
+    a->sort);
+}
+
+smt_astt bitwuzla_convt::mk_select(smt_astt a, smt_astt b)
+{
+  assert(a->sort->id == SMT_SORT_ARRAY);
+  assert(a->sort->get_domain_width() == b->sort->get_data_width());
+  return new_ast(
+    bitwuzla_mk_term2(
+      bitw,
+      BITWUZLA_KIND_ARRAY_SELECT,
+      to_solver_smt_ast<bitw_smt_ast>(a)->a,
+      to_solver_smt_ast<bitw_smt_ast>(b)->a),
+    a->sort->get_range_sort());
+}
+
+smt_astt bitwuzla_convt::mk_smt_int(const BigInt &theint [[gnu::unused]])
+{
+  std::cerr << "Bitwuzla can't create integer sorts" << std::endl;
+  abort();
+}
+
+smt_astt bitwuzla_convt::mk_smt_real(const std::string &str [[gnu::unused]])
+{
+  std::cerr << "Bitwuzla can't create Real sorts" << std::endl;
+  abort();
+}
+
+smt_astt bitwuzla_convt::mk_smt_bv(const BigInt &theint, smt_sortt s)
+{
+  // Fedor: double check the second argument of the function
+  BitwuzlaSort *bitwsort = bitwuzla_mk_bv_sort(bitw, s->get_data_width());
```

Comment:
you don't need to create another sort here, it is already inside the smt_sortt wrapper.